### ISSUE TITLE
Spend limit permission not recognized

### ISSUE THREAD

**Issue by sacalo** [curvefi/curve-contract#137]
It seems that the spending allowance is not recognized if it's already done before starting the exchange.

This is what I did:
I wanted to exchange aUSDT for aUSDC. 
When I clicked "Sell" button, metamask asked me to approve spend limit. I edited the amount to a 1% more than I wanted to exchange. At the next step I saw a fee of 50usd, so I cancelled  the trade, but the spending allowance is already confirmed.
Now, if I try to start again the process, it asks again for the spending limit aproval, even though it's already approved.

**Comment by MODERATOR10** [comment]
Hello @sacalo have you been able to exchange your aUSDT for aUSDC yet ?
